solr/solrj/src/java/org/apache/solr/client/solrj/request/JavaBinUpdateRequestCodec.java

```diff
@@ -114,65 +98,48 @@ public void marshal(UpdateRequest updateRequest, OutputStream os) throws IOExcep
   @SuppressWarnings({"unchecked"})
   public UpdateRequest unmarshal(InputStream is, final StreamingUpdateHandler handler)
       throws IOException {
-    final UpdateRequest updateRequest = new UpdateRequest();
-    List<List<NamedList<?>>> doclist;
-    List<Entry<SolrInputDocument, Map<Object, Object>>> docMap;
-    List<String> delById;
-    Map<String, Map<String, Object>> delByIdMap;
-    List<String> delByQ;
-    final NamedList<?>[] namedList = new NamedList<?>[1];
-    try (JavaBinCodec codec = new StreamingCodec(namedList, updateRequest, handler)) {
-      codec.unmarshal(is);
+    final NamedList<Object> namedList;
+
+    // process documents:
+
+    // reads documents, sending to handler.  Other data is in NamedList
+    try (var codec = new StreamingCodec(handler)) {
+      namedList = codec.unmarshal(is);
```

Comment:
big improvement, simplifying the request/response contract here with the internal StreamingCodec.  No need to have the array of NamedList or pass that in or pass an UpdateRequest.  Now a simple NamedList response.